code/datums/cinematic.dm

```diff
@@ -0,0 +1,223 @@
+GLOBAL_LIST_EMPTY(cinematics)
+
+// Use to play cinematics.
+// Watcher can be world,mob, or a list of mobs
+// Blocks until sequence is done.
+/proc/Cinematic(id,watcher,datum/callback/special_callback)
+	var/datum/cinematic/playing
+	for(var/V in subtypesof(/datum/cinematic))
+		var/datum/cinematic/C = V
+		if(initial(C.id) == id)
+			playing = new V()
+			break
+	if(!playing)
+		CRASH("Cinematic type not found")
+	if(special_callback)
+		playing.special_callback = special_callback
+	if(watcher == world)
+		playing.is_global = TRUE
+		watcher = GLOB.mob_list
+	playing.play(watcher)
+
+/obj/screen/cinematic
+	icon = 'icons/effects/station_explosion.dmi'
+	icon_state = "station_intact"
+	layer = 21
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	screen_loc = "1,1"
+
+/datum/cinematic
+	var/id = CINEMATIC_DEFAULT
+	var/list/watching = list() //List of clients watching this
+	var/list/locked = list() //Who had notransform set during the cinematic
+	var/is_global = FALSE //Global cinematics will override mob-specific ones
+	var/obj/screen/cinematic/screen
+	var/datum/callback/special_callback //For special effects synced with animation (explosions after the countdown etc)
+	var/cleanup_time = 300 //How long for the final screen to remain
+
+/datum/cinematic/New()
+	GLOB.cinematics += src
+	screen = new(src)
+
+/datum/cinematic/Destroy()
+	GLOB.cinematics -= src
+	qdel(screen)
+	for(var/mob/M in locked)
+		M.notransform = FALSE
+	return ..()
+
+/datum/cinematic/proc/play(watchers)
+	//Check if you can actually play it (stop mob cinematics for global ones) and create screen objects
+	for(var/A in GLOB.cinematics)
+		var/datum/cinematic/C = A
+		if(C == src)
+			continue 
+		if(C.is_global || !is_global)
+			return //Can't play two global or local cinematics at the same time
+
+	//Close all open windows if global
+	if(is_global)
+		for (var/datum/html_interface/hi in GLOB.html_interfaces)
+			hi.closeAll()
+		SStgui.close_all_uis()
+
+
+	for(var/mob/M in GLOB.mob_list)
```

Comment:
us player_list instead

Reply:
no_transform still needs to be set on all for global ones so this saves a second go over.

Reply:
Then again, the whole no-transform is kinda debatable anyway.